File: test-suite-python/src/test/python/micronaut/docs/openapi/informer/example2/CustomTypeNameMapper.py
from jakarta.inject import Singleton
from micronaut.kubernetes.client.openapi.watcher.mapper import TypeNameMapper


# tag::get[]
@Singleton
class CustomTypeNameMapper(TypeNameMapper):
    def getMappings(self) -> dict[str, str]:
        return {
            "micronaut.docs.openapi.informer.example2.CustomObjectCollection": "micronaut.docs.openapi.informer.example2.CustomObject"
        }
# end::get[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/openapi/informer/example2/CustomTypeNameMapper.kt
package io.micronaut.docs.openapi.informer.example2

import io.micronaut.kubernetes.client.openapi.watcher.mapper.TypeNameMapper
import jakarta.inject.Singleton

//tag::get[]
@Singleton
class CustomTypeNameMapper : TypeNameMapper {
    override fun getMappings(): Map<String, String> {
        val mappings = HashMap<String, String>()
        mappings["io.micronaut.docs.openapi.informer.example2.CustomObjectCollection"] = "io.micronaut.docs.openapi.informer.example2.CustomObject"
        return mappings
    }
}
//end::get[]
